src/shared/lib/HotspotSet.ts

```diff
@@ -0,0 +1,95 @@
+/* HotspotSet
+ *
+ * This class is designed for quickly querying whether a mutation is a hotspot.
+ * It's supported to query by a single residue position, or a range of residue positions,
+ *  and it does a binary search to quickly check whether the input range or position
+ *  overlaps with any hotspot regions specified in the constructor.
+ *
+ * The class does not support telling you exactly which input intervals are overlapped-with,
+ *  that information is lost. But it will tell you true or false: whether there was some input interval
+ *  that is overlapped-with.
+ */
+export default class HotspotSet {
+    private regions:[number,number][];
+    constructor(intervals:[number,number][]) {
```

Comment:
how about defining a type for interval? Something like this:
```JavaScript
type Interval {
    start: number,
    end: number
}
```

And then define `regions` and `intervals` like this:

```JavaScript
    regions: Interval[]
    intervals: Interval[]
```